xchange-binance/src/main/java/info/bitrich/xchangestream/binance/BinanceStreamingExchange.java

```diff
@@ -165,16 +163,34 @@ private BinanceStreamingService createStreamingService(ProductSubscription subsc
         return new BinanceStreamingService(path, subscription);
     }
 
-    public static String buildSubscriptionStreams(ProductSubscription subscription) {
+    public String buildSubscriptionStreams(ProductSubscription subscription) {
         return Stream.of(buildSubscriptionStrings(subscription.getTicker(), "ticker"),
                 buildSubscriptionStrings(subscription.getOrderBook(), "depth"),
                 buildSubscriptionStrings(subscription.getTrades(), "trade"))
                     .filter(s -> !s.isEmpty())
                     .collect(Collectors.joining("/"));
     }
 
-    private static String buildSubscriptionStrings(List<CurrencyPair> currencyPairs, String subscriptionType) {
-        return subscriptionStrings(currencyPairs).map( s -> s + "@" + subscriptionType).collect(Collectors.joining("/"));
+    private String buildSubscriptionStrings(List<CurrencyPair> currencyPairs, String subscriptionType) {
+        if ("depth".equals(subscriptionType)) {
+            Integer updateFrequency = (Integer) exchangeSpecification
```

Comment:
If you simply had a `private int orderBookUpdateFrequency` extracted in `initServices`, which defaults to 1000ms, all this code would simplify down to a single line. You wouldn't need any `if` conditions at all.

Alternatively, if you want to default to whatever Binance provide by default, you could make `orderBookUpdateFrequency` a `String` which defaults to `""`.